pkg/controller/modeladapter/modeladapter_controller.go

```diff
@@ -343,72 +342,41 @@ func (r *ModelAdapterReconciler) DoReconcile(ctx context.Context, req ctrl.Reque
-		// TODO: this needs to be changed once we support multiple lora adapters
-		selectedPodName := instance.Status.Instances[0]
-		if err := r.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: selectedPodName}, selectedPod); err != nil && apierrors.IsNotFound(err) {
-			klog.ErrorS(err, "Selected pod has been deleted and it should be removed from model adapter instance list", "modelAdapter", klog.KObj(instance))
-			// instance.Status.Instances has been outdated, and we need to clear the pod list
-			// after the pod list is cleaned up, let's reconcile the instance object again in the next loop
-			return ctrl.Result{}, r.clearModelAdapterInstanceList(ctx, instance, selectedPodName)
-		} else if err != nil {
-			// failed to fetch the pod, let's requeue
-			return ctrl.Result{RequeueAfter: defaultRequeueDuration}, err
-		} else {
-			// compare instance and model adapter labels.
-			selector, err := metav1.LabelSelectorAsSelector(instance.Spec.PodSelector)
-			if err != nil {
-				// TODO: this should barely happen, let's move this logic to earlier validation logics.
-				return ctrl.Result{}, fmt.Errorf("failed to convert pod selector: %v", err)
-			}
-			if !selector.Matches(labels.Set(selectedPod.Labels)) {
-				klog.Warning("current assigned pod selector doesn't match model adapter selector")
-				return ctrl.Result{}, r.clearModelAdapterInstanceList(ctx, instance, selectedPodName)
-			}
+	// Step 1: Sync Pod instances for ModelAdapter
+	activePods, err := r.getActivePodsForModelAdapter(ctx, instance)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
 
-			// base model pod could be unhealthy or in termination, let's clean up the instances.
-			if !utils.IsPodReady(selectedPod) || utils.IsPodTerminating(selectedPod) {
-				klog.Warningf("current assigned pod %s/%s is not ready, remove it and reschedule the adapter", selectedPod.Namespace, selectedPod.Name)
-				return ctrl.Result{}, r.clearModelAdapterInstanceList(ctx, instance, selectedPodName)
-			}
+	activeMap := make(map[string]corev1.Pod, len(activePods))
+	for _, p := range activePods {
+		activeMap[p.Name] = p
+	}
 
-			existPods = true
+	var updatedInstances []string
+	for _, name := range instance.Status.Instances {
+		if _, ok := activeMap[name]; ok {
+			updatedInstances = append(updatedInstances, name)
 		}
 	}
+	instance.Status.Instances = updatedInstances
 
-	if !existPods {
-		// TODO: as we plan to support lora replicas, it needs some corresponding changes.
-		// it should return a list of pods in future, otherwise, it should be invoked by N times.
-		activePods, err := r.getActivePodsForModelAdapter(ctx, instance)
-		if err != nil {
-			return ctrl.Result{}, err
+	added := false
+	for name := range activeMap {
+		if !StringInSlice(instance.Status.Instances, name) {
+			instance.Status.Instances = append(instance.Status.Instances, name)
+			added = true
 		}
-		if len(activePods) != 0 {
-			selectedPod, err = r.schedulePod(ctx, instance, activePods)
```

Comment:
em. @dittops can you confirm the scheduling logic has been disabled in the new workflow? any reasons?